src/HackAnalyzer.java

```diff
@@ -0,0 +1,33 @@
+class HackAnalyzer {
+    private static final String HACK_PATTERN_1 = ". ";
+    private static final String HACK_PATTERN_2 = ", ";
+    private static final String HACK_PATTERN_3 = "\\w\\.\\w";
+    private static final String HACK_PATTERN_4 = "\\.\\w\\.";
+
+
+    //return -1 if there are no matches
+    //return 0 if there is one match
+    //return 1 if there are more than one matches
+    static int analyzeForComplianceWithTheMainPatterns(StringBuilder analyzedText){
+        int count = -1;
```

Comment:
there's no need for assigning -1 to count in here as well as this variable's gonna be reassigned anyways

Reply:
y, its true